zlib-rs/src/deflate/algorithm/quick.rs

```diff
@@ -35,7 +25,7 @@ pub fn deflate_quick(stream: &mut DeflateStream, flush: DeflateFlush) -> BlockSt
                     state = &mut stream.state;
                 }
                 if stream.avail_out == 0 {
-                    return match last {
+                    return match $last {
```

Comment:
this was the actual issue, `$last` was captured instead of the macro argument being used. I've moved the macros above the definition of `last` now so that this cannot happen by accident again.